### PR TITLE
Modernize monorepo structure

### PR DESCRIPTION
**Pull request by luiscubal**
Changes:

  * Migrate from old `lerna` to `pnpm`, which is better supported and generally works better.
  * Configure commit linting
  * Configure basic Eslint
  * Configure changelogs
  * Update babel config, to only support node >= 12

TODO:

  * Fix eslint warnings: I think this doesn't belong in this PR.
  * Setup pipelines: I also think this should be a separate PR, as these changes are large enough as they are.

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+pnpm changeset version
+pnpm install
+git add .changeset pnpm-lock.yaml
+git commit -m "chore: release new version"
+git push
+pnpm publish -r
```
Comment (vitorveiga) [inline scripts/release.sh:10]: should we use the publish command of changesets?

**Comment by vitorveiga** [comment]
Can we add this bot https://github.com/apps/changeset-bot?

**Comment by luiscubal** [comment]
> Can we add this bot https://github.com/apps/changeset-bot?

Done.

File: package.json [hunk 2]
```diff
@@ -1,5 +1,44 @@
...
+  "keywords": [],
+  "author": "Jscrambler <support@jscrambler.com>",
+  "license": "MIT",
+  "engines": {
+    "pnpm": "^8.14.0"
+  },
+  "scripts": {
+    "preinstall": "only-allow pnpm",
+    "postinstall": "husky install",
+    "commit": "git-cz",
```
[the remark this answers is not in the document]
Comment (luiscubal) [reply package.json:14]: Removed.

File: scripts/release.sh (hunk 1 above)
Comment (luiscubal) [reply scripts/release.sh:10]: They are identical: https://github.com/changesets/changesets/blob/main/docs/command-line-options.md

> If you are using pnpm as a package manager, this automatically detects it and uses pnpm publish instead.

**Pull request merged**